src/core/ddsi/src/ddsi_proxy_participant.c

```diff
@@ -534,7 +540,8 @@ void ddsi_unref_proxy_participant (struct ddsi_proxy_participant *proxypp, struc
 #ifdef DDS_HAS_TOPIC_DISCOVERY
     /* Last unref is called from gc_delete_proxy_participant, which is added to the gc queue after all proxy
        topic gc tasks. So we can safely assert that the proxy topic list is empty at this point */
-    assert (proxy_topic_list_count (&proxypp->topics) == 0);
+    assert (ddsrt_avl_is_empty (&proxypp->topics));
+    ddsrt_avl_free (&ddsi_proxypp_proxytp_treedef, &proxypp->topics, 0);
```

Comment:
This call to `ddsrt_avl_free` is not really going to do anything given that it is empty — but it is also not wrong, and it is not worth the effort of removing it.